### ISSUE TITLE
Issues with keepRenderingSort

### ISSUE THREAD

**Issue by rmatherly** [crlcu/multiselect#25]
I am using the keepRenderingSort option and both lists have values at startup.  This leads to somewhat sporadic behavior when moving items between the lists.  It seems to add the moved item to the top or bottom of the list and the sort order flips from ascending to descending each time.

I'm not sure what is going on with the sort order flipping, but I have an idea of how to fix the issue with position.  If you can assign position values to the items in the Left list, which starts out with all the values, then as you compare it to the right list, copy the position value to the matching items in the Right list before you remove them from the Left list.  That way, as you move items back and forth, they will always have the correct position as defined by the original list.

If I get a chance, I will try to fix it and post the code here, but if you are able to take a look or provide any insight, that would be helpful.

**Comment by crlcu** [comment]
Hi @rmatherly 
I just had a look at the issue you described.

It is not necessary to have the items from the ``right`` side into the ``left`` side when you render the page - but if you have them in both, what is in ``right`` side it will be removed from ``left`` side.

As a quick fix - you can maybe no put all the values to the ``left`` and only selected ones to the ``right``. When you render the page you can somehow deal with this.

Now if we are talking about the situation when you have items in both sides when the page is rendered (what is in ``left`` is only in ``left``), it is a bit tricky to decide which is the ``rendered`` order - that's why I'm assigning ``data-position`` for items from both ``select`` elements - which leads having items with same position when you move from ``left`` to ``right`` and vice versa.

I will try to find a way of having a better approach for this kind of situation, but now I have no idea which is the best solution.

**Comment by rmatherly** [comment]
Hi Adrian,
Thanks for taking the time to look at this.  It seems that the automatically assigned data-position for rendered order only works correctly when the right box is empty at the start.  For other cases, would it would make sense to have a manual option where the developer would populate the data-position attribute on all the option elements?  That would make the developer responsible for assigning the correct sort values to the options regardless of which box they are in.

Thanks,
Rick Matherly

**Comment by wlrichards** [comment]
Hi Adrian,

I have encountered the same problem. The issue is with the position based sorting used with KeepRenderingSort. This sorting kind of works if you always start with an empty right list. However if the right list has items when the plugin is installed then they get assigned their own relative positions which look wrong when they are moved to the left list. Likewise moving an item from right to left looks weird if it suddenly pops in above the existing items.

For my use case the order that items are added to the right list is significant and should be preserved. I have implemented my own Move Up/Down buttons as well for this list.

I have two suggestions for fixing this and I can send you a pull request for either if you like.

In an ideal world I would like to see the following behaviour for  KeepRenderingSort:
- Items moved from left to right are always inserted at the end of the right list.
- Items move from right to left snap back into their original position
- Positions in the left list are not set by the plugin if they have already been preset in the HTML

The last item would allow me to assign absolute positions for items from my server side code so that when the right list is pre-filled on plugin startup it knows what the correct position is for items moved right to left.

The second option is to have a third sorting type which is "no sorting". Items moved either way would simply be added to the end of the list. I would then use the afterMoveToLeft event to do my own sorting.

Thanks in advance
Wendy
